lib/slave/src/FSM.cc

```diff
@@ -0,0 +1,53 @@
+#include "kickcat/FSM.h"
+#include "AbstractESC2.h"
+
+namespace kickcat::FSM
+{
+    AbstractState::AbstractState(kickcat::AbstractESC2& esc, std::string const& name)
+        : name_{name}
+        , esc_{esc}
+    {
+    }
+
+    std::string const& AbstractState::name()
+    {
+        return name_;
+    }
+
+    void AbstractState::onEntry(uint8_t) {};
+
+    void AbstractState::onExit(uint8_t) {};
+
+    StateMachine::StateMachine(std::map<uint8_t, FSM::AbstractState*>& states, kickcat::State defaultState)
+        : states_{states}
+    {
+        currentState_ = defaultState;
+        defaultState_ = defaultState;
+    }
+
+    void StateMachine::init()
```

Comment:
start() seems to be a better name (since it is starting working on states at this point)